pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -32,10 +34,58 @@ def __init__(self, runtime: Runtime, version: str, assembly: str, publish: bool,
         self.images = images
         self.exclude_arches = [] if not exclude_arches else exclude_arches.replace(',', ' ').split()
         self.skip_multiarch_payload = skip_multiarch_payload
+        self.triggered_from_gen_assembly = triggered_from_gen_assembly
+        self.build_number = build_number
         self.logger = runtime.logger
         self.working_dir = self.runtime.working_dir
 
+    async def comment_on_pr(self):
+        """
+        Comment the link to this jenkins build on the assembly PR if it was triggered automatically
+        """
+        if self.triggered_from_gen_assembly:
```

Comment:
If `comment_on_pr` is called with `self.triggered_from_gen_assembly=False`, `comment_on_pr` will do nothing. This sounds confusing.
Please move the `if self.triggered_from_gen_assembly:` conditional check out of this function.

Reply:
Understood! Updated!